packages/ics721/src/execute.rs

```diff
@@ -164,6 +164,9 @@ where
             return Err(ContractError::Unauthorized {});
         }
 
+        // cw721 doesn't support on-chain metadata yet
+        // here NFT is transferred to another chain, NFT itself may have been transferred to his chain before
+        // in this case ICS721 may have metadata stored
         let token_metadata = TOKEN_METADATA
             .may_load(deps.storage, (class.id.clone(), token_id.clone()))?
```

Comment:
during `receive_nft()` (starting point from nft contract to ics721 for transfer) read  TOKEN_METADATA (in case it has been provided on previous, initial transfer to this chain), this is used for transferring to next chain. Not part of this PR. just outlining what happens here.